### PR TITLE
32 bit TI (eliminate 32767 record limit)

### PR DESCRIPTION
**Pull request by chipbarnaby**
## Description

Change TI from 16 bit (int16_t) to 32 bit (int32_t).  This allows CSE to have more than 32767 records in a given RAT.

No results changes.  No documentation changes.

Note: modified code does not detect / handle other memory limits.  See issue 502.

### CODE REVIEW

File: src/ancrec.h [hunk 1]
```diff
@@ -457,13 +459,15 @@ x    n = nAl = 0;				// say no records in use, none allocated: insurance
...
-template <class T> void anc<T>::desRecs( SI _mn, SI _n)
+#if 0
+template <class T> void anc<T>::desRecs( TI _mn, TI _n)
 // ?? why isn't basAnc::desRecs sufficient, 5-31-92?
 {
     if (ptr()) 						// if record memory is allocated
        for (TI i = max(mn,_mn);  i <= min(n,_n);  i++) 	// loop allocated record spaces in range given by caller
           desRec(i);					// conditionally destroy record in space with record-deriv d'tor ~T.
 }			// anc<T>::desRecs
+#endif
```
Comment (nealkruis) [inline src/ancrec.h:470]: Keep or remove?

File: src/ancrec.h [hunk 2]
```diff
@@ -373,7 +375,7 @@ template <class T>  class anc : public basAnc
 
 
  protected:
-    void desRecs( TI mn=0, TI n=32767); 			// ~ all records or range (as b4 freeing block)
+    // void desRecs( TI mn=0, TI n=32767); 			// ~ all records or range (as b4 freeing block)
```
Comment (nealkruis) [inline src/ancrec.h:378]: Keep or remove?

**Review by nealkruis — CHANGES_REQUESTED** [review]
Looks fine. Just need to clean up temporary code.

File: src/ancrec.h (hunk 2 above)
Comment (chipbarnaby) [reply src/ancrec.h:378]: Remove

File: src/ancrec.h (hunk 1 above)
Comment (chipbarnaby) [reply src/ancrec.h:470]: Remove

**Review by nealkruis — APPROVED** [review]

**Pull request merged**